graph/pull.go

```diff
@@ -463,6 +469,9 @@ func (s *TagStore) pullV2Tag(eng *engine.Engine, r *registry.Session, out io.Wri
 		}
 		downloads[i].img = img
 
+		layers = append(layers, img.ID)
+		s.graph.Retain(layers, manifest.Name)
```

Comment:
Every time this is called, won't the layers which were previously added keep appending the same name to the retain list?  Based on the code it doesn't appear this would lead to incorrect behavior but just doesn't seem like the cleanest way to do it. Maybe just `s.graph.Retain([]string{img.ID}, manifest.Name)` is needed?